src/js/export.js

```diff
@@ -0,0 +1,238 @@
+(function (){
+  "use strict";
+
+  window.export = {
+    toFile:exportCollectionToFile
+  }
+
+  var cID;
+  var cName;
+  var filename;
+  var CSVDelimiter;
+  var leafDelimiter;
+
+  var delimiters = [
+      {value: ',' , display: 'Comma (,)'},
+      {value: ';' , display: 'Semi-Colon (;)'},
+      {value: ':' , display: 'Colon (:)'},
+      {value: '|' , display: 'Pipe (|)'},
+      {value: '^' , display: 'Caret (^)'},
+      {value: '~' , display: 'Tilde (~)'},
+      {value: '\t' , display: 'Tab'},
+      {value: ' ' , display: 'Space'}
+  ];
+
+  function exportCollectionToFile(collID, colName){
+    cID = collID;
+    cName = colName;
+    var modal = createExportModal();
+
+    // Comma should probably not be the default delimiter
+    // since a lot of the entries contain commas.
+    selectDelimiterCSV.selectedIndex = 1;
+    selectDelimiterLeaf.selectedIndex = 3;
+
+    closeBtn.addEventListener('click', function() {
+      document.body.removeChild(modal);
+    }, false);
+
+    cancelBtn.addEventListener('click', function() {
+      document.body.removeChild(modal);
+    }, false);
+
+    exportBtn.addEventListener('click', (evt) => {
+      clearComplaintsExport();
+      filename = document.getElementById("exportFilename").value;
+      var filenameValid = isFilenameValid();
+      var delimitersValid = areDelimitersValid();
+      if(!filenameValid || !delimitersValid){
```

Comment:
@smeetz Do you mean 'if(!isFilenameValid() || !areDelimitersValid())' ?

The reason the code is not written like that is because if the first function returns false, the second function will not be run (according to how || works). Both of these functions will append a complaint next to their field if the input is incorrect. So if the first function returns false the second function will not append the complaint, even if the input for that function is incorrect.